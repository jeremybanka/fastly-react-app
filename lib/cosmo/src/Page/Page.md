```jsx
<Page>
  <Page.Header>
    <Page.Title>Page Title</Page.Title>
    <Page.Subtitle>Page Subtitle</Page.Subtitle>
  </Page.Header>
  <Page.Body>PageBody content</Page.Body>
  <Page.Footer>PageFooter content</Page.Footer>
</Page>
```
