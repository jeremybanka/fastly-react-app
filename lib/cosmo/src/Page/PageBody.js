/* @flow */

import * as React from "react";
import { Box } from "../";

type Props = { children: React.Node };

const PageBody = (props: Props) => (
  <Box marginBottom="lg">{props.children}</Box>
);

export default PageBody;
