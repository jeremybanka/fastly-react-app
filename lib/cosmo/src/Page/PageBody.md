The `<Page.Body />` component should be nested inside of a `<Page />`.
