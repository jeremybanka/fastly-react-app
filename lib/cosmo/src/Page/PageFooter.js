/* @flow */

import * as React from "react";
import { Box } from "../";

type Props = { children: React.Node };

const PageFooter = (props: Props) => (
  <Box paddingTop="md" marginTop="lg">
    {props.children}
  </Box>
);

export default PageFooter;
