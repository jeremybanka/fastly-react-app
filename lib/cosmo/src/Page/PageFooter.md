The `<Page.Footer />` component should be nested inside of a `<Page />`.
