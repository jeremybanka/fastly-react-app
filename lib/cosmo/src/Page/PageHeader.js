/* @flow */

import * as React from "react";
import { Box } from "../";

type Props = { children: React.Node };

const PageHeader = (props: Props) => (
  <Box paddingBottom="md" marginBottom="lg" borderBottom="regular">
    {props.children}
  </Box>
);

export default PageHeader;
