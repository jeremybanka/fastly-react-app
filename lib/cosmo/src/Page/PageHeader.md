The `<Page.Header />` component should be nested inside of a `<Page />`.
