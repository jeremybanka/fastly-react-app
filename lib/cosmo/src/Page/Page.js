/* @flow */

import * as React from "react";
import { Box } from "../";
import PageBody from "./PageBody";
import PageFooter from "./PageFooter";
import PageHeader from "./PageHeader";
import PageTitle from "./PageTitle";
import PageSubtitle from "./PageSubtitle";

type Props = { children: React.Node };

const Page = (props: Props) => {
  return (
    <Box position="relative" marginTop="lg">
      {props.children}
    </Box>
  );
};

Page.Body = PageBody;
Page.Footer = PageFooter;
Page.Header = PageHeader;
Page.Title = PageTitle;
Page.Subtitle = PageSubtitle;

export default Page;
