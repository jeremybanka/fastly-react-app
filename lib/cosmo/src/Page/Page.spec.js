/* @flow */

import React from "react";
import { shallow, render } from "enzyme";

import Page from "./Page";

describe("Page test suite", () => {
  it("renders with className .Page", () => {
    expect(shallow(<Page />).is(".Page")).toBe(true);
  });

  it("renders the stored snapshot", () => {
    expect(render(<Page />)).toMatchSnapshot();
  });
});
