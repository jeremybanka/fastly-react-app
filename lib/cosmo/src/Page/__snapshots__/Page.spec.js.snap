// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page test suite renders the stored snapshot 1`] = `
<div
  class="Page"
/>
`;
